File: shared/src/types/daily-checkin.ts
export type CheckinType = "daily" | "post_activity" | "pre_visit";

export interface DailyCheckinRecord {
  id: string;
  client_id: string;
  clinic_id: string;
  checkin_type: CheckinType;
  overall_feeling: number;
  target_regions: Array<Record<string, unknown>>;
  activity_since_last: string | null;
  recovery_score: number | null;
  created_at: string;
  updated_at?: string;
}

File: HydraScan/HydraScan/Models/DailyCheckin.swift
import Foundation

enum CheckinType: String, Codable, CaseIterable, Hashable, Identifiable {
    case daily
    case postActivity = "post_activity"
    case preVisit = "pre_visit"

    var id: String { rawValue }
}

struct DailyCheckin: Identifiable, Codable, Hashable {
    var id: UUID
    var clientID: UUID
    var clinicID: UUID?
    var checkinType: CheckinType
    var overallFeeling: Int
    var targetRegions: [BodyRegion]
    var activitySinceLast: String?
    var recoveryScore: Double?
    var createdAt: Date
    var updatedAt: Date?
}
